### DISCUSSION TITLE
How can I retrieve all users that were inactive since date?

### DISCUSSION THREAD

**Discussion by moonvader** [spatie/laravel-activitylog#727] [Q&A] [answered]
If I use laravel-activitylog how can I get list of inactive users for period of dates?

**Comment by Gummibeer** [comment, marked as answer]
If your user model uses this trait https://github.com/spatie/laravel-activitylog/blob/master/src/Traits/CausesActivity.php you can use the `actions()` relationship and use the `created_at` column.
```php
$avtiveUsers = User::whereHas('actions', fn(Builder $q) => $q->where('created_at', '>=', now()->subWeek()))->get();
```
This will return all users with at least one activity in the last week. You can invert the `whereHas()` to get all inactive or change the carbon date to adjust the period to check.
It's a normal eloquent relationship so you can do whatever you want on the query.
